src/goto-symex/symex_clean_expr.cpp

```diff
@@ -166,12 +168,41 @@ void goto_symext::replace_array_equal(exprt &expr)
     replace_array_equal(*it);
 }
 
+/// Rewrite index/member expressions in byte_extract to offset
+static void adjust_byte_extract_rec(exprt &expr, const namespacet &ns)
+{
```

Comment:
Wouldn't the natural place for this to happen be the simplifier?

Reply:
While that is true, it would cause/require repeated simplification of the same expression (note that this simplification needs to happen before doing `symex_assign_*`). It may, however, also be useful to add this to the simplifier as a fallback?

Reply:
Not sure the object descriptors fit into simplifier; leave as is for now.